src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityKMSRequest.java

```diff
@@ -58,6 +59,7 @@
         this.batchUnwrapEndpoint = new GenericUrl(tspApiPrefix + "document/batch-unwrap");
 
         this.webRequestExecutor = Executors.newFixedThreadPool(requestThreadSize);
+        this.requestFactory = provideHttpRequestFactory(requestThreadSize, requestThreadSize);
```

Comment:
is this really what we want? The pool is the size of requestThreadSize, but if they're calling both wrap and unwrap shouldn't the route pool be smaller? (Assuming route pool size actually means per route)

Reply:
Without introducing more config/construction knobs I think what I have is the best we can do. If two routes are being called, making the route pool smaller would be more _fair_, but this would come at the cost of if only one route is being called, the number of concurrent requests will be artificially limited to a smaller number for no reason. 

If we want to make more knobs I think we can make this better. With one knob, I think this is fine.